src/Utils.ts

```diff
@@ -16,17 +16,24 @@ export function fromEditorConfig(
 ): TextEditorOptions {
 	const resolved: TextEditorOptions = {
 		tabSize: (config.indent_style === 'tab'
+			|| config.indent_size === 'tab'
 			? get(config, 'tab_width', config.indent_size)
-			: get(config, 'indent_size', config.tab_width)
-		)
+			: config.indent_style === 'space'
+			? get(config, 'indent_size', config.tab_width)
+			: get(config, 'indent_size', config.tab_width, defaults.tabSize)
+		),
+		insertSpaces: config.indent_style
+			? config.indent_style === 'space'
+			: defaults.insertSpaces
 	};
-	if (get(resolved, 'tabSize') === 'tab') {
-		resolved.tabSize = config.tab_width;
+	if ((!config.indent_style || config.indent_style === 'unset')
+		&& config.indent_size !== 'tab'
+		&& defaults && Object.keys(defaults).length === 0) {
+			resolved.tabSize = 'auto';
```

Comment:
It looks like vscode's default `tabSize` is `4`, which is overridden by the `editor.detectIndentation` setting.
```json
// The number of spaces a tab is equal to.
// This setting is overridden based on the file contents
// when `editor.detectIndentation` is on.
  "editor.tabSize": 4,
```
I think it's best that we don't set these values to `auto`, as that's not the default, built-in value. In fact, I'm not sure we have to set it to anything if we want to get the default, built-in value.